rfcs/react-components/convergence/long-term-support-plan.md

```diff
@@ -0,0 +1,133 @@
+
+This plan requires communicating to partners that own their own v0 and v8 based packages in the master branch that they will need to start working on them in the new branches.
+
+There should be a cut-over date where branches are locked for changes and then the steps taken (scripts?) to make the change followed by the branches being unlocked. It would be good to time
+the cutover to a time of low activity for all teams globally.
+
+<sup>1</sup> Multiple migration partners have asked 'where is the v9 code?', 'are packages at the same level as react or react-components v8 or v9?', and 'do improvements in react or react-components get cherry-picked to the other folder?'. We've had similar questions come up on our shield and 1JS channels.
+
+<sup>2</sup> Includes branch protection policies.
+
+<sup>3</sup> react-portal-compat, react-portal-compat-context
+
+<sup>4</sup> react-migration-v0-v9, react-migration-v8-v9
+
+<sup>5</sup> Today our builds have some conditional code specific to v8 or v9 builds. This work would remove those conditions.
+
+#### Flatten master branch
+
+With northstar-v0 and v8 no longer in the master branch, we can reduce the depth of the tree by bringing components within the react-components folder up one level.
+
+#### Sunset v8
+
+After retiring v7, announce v8 is in sunset mode. Only critical bugs will be fixed. We will no longer be moving React forward in v8. Set a retirement date for one year out.
+
+#### Define team tenets for active versions
+
+The Fluent team size doesn't allow for supporting more than 2 versions of Fluent. We should consider policy where we don't start active development on a new major version until the oldest version is retired (or at least in sunset). We can have a next branch and do lots of experimentation there, but there would be no support for this branch.
+
+Branch renames make it possible to have an easier current verson change if we do next version development it its own branch. We can rename master to vX and then rename next to master.
+
+#### Consider renaming master to main
+
+In 2020, as part of replacing exclusionary terms with more inclusive ones, many repositories started renaming the master branch to main. New repositories in GitHub now default to main. We should take this opportunity to do the same if we are renaming branches.
+
+### Pros and Cons
+
+- 👍 v7 is already isolated in own branch, requiring no changes and will be familiar to partners
+- 👍 branches provide independent policy allowing for easier management
+- 👍 folder depth is reduced and packages more discoverable
+- 👍 build times should be moderately faster with no if/then logic<sub>1<sub>
+- 👍 builds should be more reliable as changes are always specific to that version
+- 👍 build technology can evolve more rapidly as there is less compatibility work to consider and reduces build script surface area.
+
+- 👎 more branches means more enlisting, github configuration, and pipeline configuration
```

Comment:
While this is a bit of a hurdle, I don't see it as a huge obstacle for contributors as long as we maintain solid documentation hygiene so people can efficiently spin up on each branch.